standard/lexical-structure.md

```diff
@@ -1488,12 +1488,12 @@ fragment PP_Line_Indicator
     ;
     
 fragment PP_Compilation_Unit_Name
-    : '"' PP_Compilation_Unit_Name_Character+ '"'
+    : '"' PP_Compilation_Unit_Name_Character* '"'
     ;
     
 fragment PP_Compilation_Unit_Name_Character
     // Any Input_Character except "
-    : ~('\u000D' | '\u000A'   | '\u0085' | '\u2028' | '\u2029' | '#')
+    : ~('\u000D' | '\u000A'   | '\u0085' | '\u2028' | '\u2029' | '"')
```

Comment:
The `#` was a typo so should be fixed…

However §22.5.6.3 defines the format of the compilation unit name/file path as “implementation-dependent”. So this section _might_ need a semantic rule saying this arbitrary string must conform to the same implementation-dependent rules §22.5.6.3, or that is does not need to (i.e. not be valid as a file path).

Reply:
Hmmm... I'm not sure. It feels like it would be okay to allow values which aren't valid filenames, when specifying the compilation unit name directly in code, even if the CallerFilePathAttribute could never automatically generate such a name. (Indeed, I can see some cases where that would even be useful!)

I think this is speaking in favor of having a semantic rule saying "that it does not need to".

Anecdotally, I observe that Roslyn is okay with this:

```csharp
#line 100 ":invalid:"
```

and even:

```csharp
#line 100 ".."
```

(Interestingly, for the latter, it reports any subsequent error as belonging to the parent directory of the directory containing the file - it doesn't report it as ".." verbatim...)

Reply:
I've only just tried code with the former (`":invalid:"`) with code containing errors after it: Roslyn crashes:

```text
error MSB6006: "csc.dll" exited with code 1.
```

Reply:
I'll report the crash to the Roslyn team.

Reply:
Also covered by #1233.